tests/testthat/test-predict.R

```diff
@@ -0,0 +1,379 @@
+test_that("predict_obs_incidence predicts observed incidence", {
+  fit <- readRDS(test_path("data", "fit.rds"))
+  expected <- data.frame(
+    reference_date = as.Date(c(
+      "2023-01-01",
+      "2023-01-02",
+      "2023-01-03",
+      "2023-01-04",
+      "2023-01-05",
+      "2023-01-06",
+      "2023-01-07",
+      "2023-01-08",
+      "2023-01-09",
+      "2023-01-10"
+    )),
+    .response = c(
+      32,
+      24,
+      46,
+      29,
+      23,
+      30,
+      15,
+      74,
+      101,
+      62
+    ),
+    .draw = c(
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L
+    )
+  )
+  actual <- predict.RtGam(fit,
+    n = 1,
+    seed = 12345,
+    parameter = "obs_incidence",
+    mean_delay = 3
+  )
+  expect_equal(actual, expected)
+})
+
+test_that("predict_obs_cases predicts observed cases", {
+  fit <- readRDS(test_path("data", "fit.rds"))
+  expected <- data.frame(
+    reference_date = as.Date(c(
+      "2023-01-01",
+      "2023-01-02",
+      "2023-01-03",
+      "2023-01-04",
+      "2023-01-05",
+      "2023-01-06",
+      "2023-01-07",
+      "2023-01-08",
+      "2023-01-09",
+      "2023-01-10"
+    )),
+    .response = c(
+      49,
+      32,
+      52,
+      30,
+      23,
+      24,
+      8,
+      34,
+      46,
+      23
+    ),
+    .draw = c(
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L,
+      1L
+    )
+  )
+  actual <- predict.RtGam(fit, n = 1, seed = 12345, parameter = "obs_cases")
```

Comment:
I'd be tempted to use predict here vs the special method as otherwise you have no test that dispatch is actually working

Reply:
https://github.com/CDCgov/cfa-gam-rt/pull/48/commits/60befeebcac5c7774a7f7c9a5045047bcb97e70c